codegen/codegen.go

```diff
@@ -666,6 +666,8 @@ func (cg *CodeGen) EmitFilesystemChainStmt(ctx context.Context, scope *parser.Sc
 				llb.Copy(input, src, dest, opts...),
 			)
 		}
+	default:
+		return nil, fmt.Errorf("%s unexpected filesystem statement: %q", checker.FormatPos(call.Pos), call.Func.Ident.Name)
```

Comment:
This should be done in the `checker` package rather than during code generation. We want to fail earlier, because in the future (with multi-targets, etc) we might have ran a few solves at this point.

Reply:
Okay, I am trying to understand the checker structure/logic now.